providers/netty4/src/main/java/org/asynchttpclient/netty/handler/Processor.java

```diff
@@ -71,11 +75,41 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) throws Exce
                 ac.call();
                 Channels.setDiscard(channel);
             }
+            ReferenceCountUtil.release(msg);
+
 
         } else if (attribute instanceof NettyResponseFuture) {
             NettyResponseFuture<?> future = (NettyResponseFuture<?>) attribute;
             protocol.handle(channel, future, msg);
 
+        } else if (attribute instanceof StreamedResponsePublisher) {
```

Comment:
If you look in the exception handling below on line 190 (the diff doesn't make it obvious that the code I added is part of the exception handling), I have handled it - the exception gets sent to the stream as an error, and then the future is sent to `requestSender.abort`.

As to whether it should be replayed or not - in Play for example, if it fails during the stream, we never want it to be replayed.  Why?  Because our API essentially looks something like this:

```java
Future<ResponseHeader, Publisher<ByteBuffer>> executeStream();
```

This future will be redeemed when `onStream` is invoked.  All subsequent calls to `onStream` will be ignored - since there is literally nothing we can do with them.  And, to imagine the most common use case for this, where you're streaming a response body out to another response body (acting essentially as a proxy), if the stream fails halfway through, what are you going to do?  You can only forward the failure to the client.  You can't retry, because your opportunity to send a response has gone.  Basically, if you're handling the response as a stream, then this very likely means that consuming the body is side effecting, in which case, it's very unlikely that you want to retry, in fact it could often be dangerous - this is a very different thing to, for example, buffering the body in memory, where there are no negative consequences to retrying.